### PR TITLE
[WIP] refactor: Breaking game data out from Application

### PR DESCRIPTION
**Pull request by Rhuagh**
Putting this out there to get feedback because it's a pretty controversial change. 

Only the core stuff is changed, none of the examples work yet.

---
This change is [Reviewable](https://reviewable.io/reviews/amethyst/amethyst/599)

### CODE REVIEW

**Comment by torkleyy** [comment]
So how is this going to work? We'll have to provide every functionality outside of the ECS, too? I thought Application was meant to be a simple wrapper, not the most customizable one.

**Comment by Rhuagh** [comment]
No, I believe we should keep using ecs, this is just to make it easier for states to share multiple dispatchers and not have to store them in the state (enum states for example are annoying to work with internal data imho).

**Comment by torkleyy** [comment]
Okay, :+1: for not requiring the dispatcher in there. Though, I don't quite understand the motivation for removing the `world` field.

File: src/app.rs [hunk 1]
```diff
@@ -333,15 +273,160 @@ impl<'a, 'b, T> ApplicationBuilder<'a, 'b, T> {
...
+}
+
+impl<'a, 'b> DefaultGameData<'a, 'b> {
+    /// Build
+    pub fn build<P: AsRef<Path>>(path: P) -> Result<DefaultGameDataBuilder<'a, 'b>> {
+        DefaultGameDataBuilder::new(path)
+    }
+
+    /// Get events
+    pub fn get_window_events<I>(&mut self) -> Vec<Event> {
```
Comment (torkleyy) [inline src/app.rs]: I'd be very much in favor of calling this `poll_window_events` since `get_` makes the impression of a simple getter.

**Comment by Rhuagh** [comment]
I didn't want to increase  the number of parameters to the state too much.

File: src/app.rs (hunk 1 above)
Comment (Rhuagh) [reply src/app.rs]: That's a better name yeah

**Comment by jojolepro** [comment]
?

**Comment by Rhuagh** [comment]
I will redo this pr, but I need some other architectural changes done first

**Pull request closed without merging**